st2common/st2common/models/utils/action_alias_utils.py

```diff
@@ -13,222 +13,71 @@
-        elif v_start < len(self._format):
-            # Advance in the format string
-            # Note: We still want to advance in the format string even though
-            # there is nothing left in the param stream since we support default
-            # values and param_stream can be empty
-            self._alias_fmt_ptr = p_end
-
-        if not value and not default_value:
-            raise content.ParseException('No value supplied and no default value found.')
-
-        return param_name, value if value else default_value
 
     def get_extracted_param_value(self):
-        result = {}
 
-        # First extract format params provided in the string (if any)
-        format_params = {name: value for name, value in self}
-        result.update(format_params)
+        result = {}
 
-        # Second extract key=value params provided in the param string
-        kv_parser = KeyValueActionAliasFormatParser(alias_format=self._format,
-                                                    param_stream=self._param_stream)
-        kv_params = kv_parser.parse()
-        result.update(kv_params)
+        # As there's a lot of questions about using regular expressions,
+        # I'll try to be thorough when documenting this code.
+
+        # We're parsing the arbitrary key-value pairs at the end of the stream
+        # to support passing of parameters not specified in the format string,
+        # and cutting them from the stream as they're no longer needed.
+        # Possible values are quoted strings, a word, or anything inside "{}".
+        pairs_match = r'(?:^|\s+)(\S+)=("(.*?)"|\'(.*?)\'|({.*?})|(\S+))'
+        extra = re.match(r'.*?((' + pairs_match + r'\s*)*)$',
+                         self._param_stream, re.DOTALL)
+        if extra:
+            kv_pairs = re.findall(pairs_match,
+                                  extra.group(1), re.DOTALL)
+            for pair in kv_pairs:
+                result[pair[0]] = ''.join(pair[2:])
+            self._param_stream = self._param_stream.replace(extra.group(1), '')
+
+        # Now we'll match parameters with default values in form of
+        # {{ value = parameter }} (and all possible permutations of spaces),
+        # compiling them into a list.
+        # "test {{ url = http://google.com }} {{ extra = Test }}" will become
+        # [ ["url", "http://google.com"], ["extra", "Test"] ]
+        params = re.findall(r'{{\s*(.+?)\s*(?:=\s*[\'"]?(.+?)[\'"]?)?\s*}}',
+                            self._format, re.DOTALL)
+
+        # Now we're transforming our format string into a regular expression,
+        # substituting {{ ... }} with regex named groups, so that param_stream
+        # matched against this expression yields a dict of params with values.
+        reg = re.sub(r'\s+{{\s*(\S+)\s*=.+?}}',
+                     r'(?:\s+[\'"]?(?P<\1>.+?)[\'"]?)?',
+                     self._format)
+        reg = re.sub(r'{{\s*(.+?)\s*}}',
+                     r'[\'"]?(?P<\1>.+?)[\'"]?',
```

Comment:
Ah, so we use backtracking here, this could become problematic with regard to pathological regex.